ddtrace/contrib/anthropic/_streaming.py

```diff
@@ -0,0 +1,322 @@
+    def __init__(self, wrapped, integration, span, args, kwargs):
+        super().__init__(wrapped)
+        n = kwargs.get("n", 1) or 1
+        self._dd_span = span
+        self._streamed_chunks = [[] for _ in range(n)]
+        self._dd_integration = integration
+        self._kwargs = kwargs
+        self._args = args
+
+
+class TracedAnthropicStream(BaseTracedAnthropicStream):
+    def __enter__(self):
+        self.__wrapped__.__enter__()
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.__wrapped__.__exit__(exc_type, exc_val, exc_tb)
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        try:
+            chunk = self.__wrapped__.__next__()
+            self._streamed_chunks.append(chunk)
+            return chunk
+        except StopIteration:
+            _process_finished_stream(
+                self._dd_integration, self._dd_span, self._args, self._kwargs, self._streamed_chunks
+            )
+            self._dd_span.finish()
+            raise
+        except Exception:
+            self._dd_span.set_exc_info(*sys.exc_info())
+            self._dd_span.finish()
+            raise
+
+    def _text_stream(self):
+        for chunk in self:
+            if chunk.type == "content_block_delta" and chunk.delta.type == "text_delta":
+                yield chunk.delta.text
+
+
+class TracedAnthropicAsyncStream(BaseTracedAnthropicStream):
+    async def __aenter__(self):
+        await self.__wrapped__.__aenter__()
+        return self
+
+    async def __aexit__(self, exc_type, exc_val, exc_tb):
+        await self.__wrapped__.__aexit__(exc_type, exc_val, exc_tb)
+
+    def __aiter__(self):
+        return self
+
+    async def __anext__(self):
+        try:
+            chunk = await self.__wrapped__.__anext__()
+            self._streamed_chunks.append(chunk)
+            return chunk
+        except StopAsyncIteration:
+            _process_finished_stream(
+                self._dd_integration,
+                self._dd_span,
+                self._args,
+                self._kwargs,
+                self._streamed_chunks,
+            )
+            self._dd_span.finish()
+            raise
+        except Exception:
+            self._dd_span.set_exc_info(*sys.exc_info())
+            self._dd_span.finish()
+            raise
+
+    async def _text_stream(self):
+        async for chunk in self:
+            if chunk.type == "content_block_delta" and chunk.delta.type == "text_delta":
+                yield chunk.delta.text
+
+
+class TracedAnthropicStreamManager(BaseTracedAnthropicStream):
+    def __enter__(self):
+        stream = self.__wrapped__.__enter__()
+        traced_stream = TracedAnthropicStream(
+            stream,
+            self._dd_integration,
+            self._dd_span,
+            self._args,
+            self._kwargs,
+        )
+        traced_stream.text_stream = traced_stream._text_stream()
```

Comment:
not sure I understand this, we aren't going to be consuming anything from the response are we?

Reply:
I removed the `traced_stream.text_stream = traced_stream._text_stream()`, but basically this stream manager class exposes a `text_stream` attribute. This attribute is actually set to a method call `text_stream=self.stream.__stream_text__`, and yields stream chunk content. It's a helper function. Instead of using `traced_stream._text_stream()`, we are just overriding `__stream_text__` directly instead.

Reply:
oh great, I like that approach, override the `__stream_text__`, makes a lot of sense!

Reply:
jk, I think we actually need to set `text_stream= our new function` since the wrapped stream holds a reference to the original function, so overriding does nothing